specs/eventing/overview.md

```diff
@@ -1,97 +1,162 @@
+agent Channel
+agent "Subscription" as S1
+agent S2 [
+Subscription
 ---
-
-_Navigation_:
-
-- [Motivation and goals](motivation.md)
-- **Resource type overview**
-- [Interface contracts](interfaces.md)
-- [Object model specification](spec.md)
+events sent from Channel to Addressable,
+replies sent to second Addressable
+]
+}
+
+Channel <-- S1 :channel
+Channel <-- S2 :channel
+Addressable <-u- S2 :subscriber
+A2 <-u- S2 : reply
+@enduml
+-->
+
+Knative Eventing does not directly specify mechanisms for other event-processing
+models, including multi-stage workflows, correlated request-reply, and
+sequential (windowed) event processing; these models could be built using the
+primitives provided by Knative, or Knative could deliver events to an external
+system that implements these models.
+
+In addition to the primitives needed to express the above patterns, Knative
+Eventing defines two [_interface contracts_](#interface-contracts) to allow
+connecting multiple types of Kubernetes objects as event senders and recipients
+to the core primitives.
+
+## Interface Contracts
+
+In addition to the concrete types described below in the `messaging.knative.dev`
+and `eventing.knative.dev` API groups, Knative Eventing supports referencing
+objects in other API groups as destinations for event delivery. This is done by
+defining partial schemas which the referenced resources must support. The
+following interface contracts define resource fragments and partial schemas
+(required fields on an arbitrary API object) which form a basis for Knative
+Eventing.
+
+### Addressable
+
+[**Addressable**](./control-plane.md#addressable-v1) resources expose a resource
+`address` (HTTP URL) in their `status` object. The URL is used as a destination
+for delivery of events to the resource; the exposed URL must implement the
+[data plane contract](data-plane.md) for receiving events.
+
+[**Broker**](#broker) and [**Channel**](#channel) both implement **Addressable**
+to receive events from other components.
+
+### Destination
+
+[**Destination**](./control-plane.md#duckv1destination) is an interface
+(resource fragment) which is used consistently through Knative Eventing to
+reference an event delivery destination. A Destination eventually resolves the
+supplied information to an URL, and may be an absolute URL or
+[relative](https://datatracker.ietf.org/doc/html/rfc3986#section-4.2) to an
+**Addressable** object reference; it also supports a Kubernetes Service object
+reference (as a special case). An absolute URL in a Destination may be used to
+reference cluster-external resources such as a virtual machine or SaaS service.
+
+### Event Source
+
+**Event Sources** are resources which generate events and may be configured to
+deliver the events to a **Destination** designated by a `sink` object in the
+resource's `spec`. The Knative Eventing spec does not define any specific event
+sources, but does define common interfaces for discovering and managing event
```

Comment:
can we turn the "discovering" word into an href so people know exactly what that refers to? Probably the same for "managing" - I'm not sure what we mean by here since I think we agreed that we don't really have much in the way of a "sources spec".

Reply:
Yeah, this was copied and aspirational. Removing.